tests/unit/unit_test_utils.cpp

```diff
@@ -46,6 +51,9 @@ VecSimQueryParams CreateQueryParams(const HNSWRuntimeParams &RuntimeParams) {
 
 static bool is_async_index(VecSimIndex *index) {
     return dynamic_cast<VecSimTieredIndex<float, float> *>(index) != nullptr ||
+           dynamic_cast<VecSimTieredIndex<bfloat16, float> *>(index) != nullptr ||
+           dynamic_cast<VecSimTieredIndex<float16, float> *>(index) != nullptr ||
```

Comment:
There is a ticket for fixing this in all relevant versions (adding the BF16 / FP16). I guess we can have it here, just remember to add it to 0.8 as well

Reply:
Hmm ok. So I’ll remove it here as we won’t BP this PR to 0.8

Reply:
Since we already changed this function on main and 8.0, we will have conflicts with 0.8 anyway. I'll keep this change and we will need to fix it on 0.8 only.